pom.xml

```diff
@@ -48,10 +48,10 @@
         <developerConnection>${scm.developer.connection}</developerConnection>
         <tag>HEAD</tag>
     </scm>
-
     <properties>
         <!-- Dependency properties -->
         <java.version>11</java.version>
+        <version.number>0.5.0</version.number>
```

Comment:
I'd also expect the project-level `<version/>` tag to be updated appropriately for this parameter:
From
```
    <version>0.5.0-${revision}</version>
```
To
```
    <version>${version.number}-${revision}</version>
```

Reply:
Would have preferred this approach, but this hits up agains an issue.  Both version and revision are reserved elements in Maven and can be used in the way that we have in the first example.  The element version.number does not exists at the point the version element is being assigned a value and so any attempt to us it fails.